aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -131,6 +131,70 @@ void index_put_kernel(TensorIterator& iter, IntArrayRef index_size, IntArrayRef
   });
 }
 
+void index_fill_kernel(
+  TensorIterator& iter,
+  int64_t dim,
+  int64_t self_dim_size,
+  int64_t self_dim_stride,
+  Scalar source) {
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(ScalarType::Half, ScalarType::Bool, ScalarType::BFloat16,
+    iter.dtype(), "index_fill_cpu", [&] {
+    auto fill_val = source.to<scalar_t>();
+    auto handle_nonzero_idx_stride = [&](char** data, const int64_t* strides, int64_t n) {
+      auto* self_data_bytes = data[0];
+      auto* index_data_bytes = data[1];
+      for (int64_t elem = 0; elem < n; ++elem) {
+        auto* self_data = reinterpret_cast<scalar_t*>(self_data_bytes);
+        auto idx = *reinterpret_cast<int64_t*>(index_data_bytes);
+        if (idx < 0) {
+          idx += self_dim_size;
+        }
+        if (idx < -self_dim_size || idx >= self_dim_size) {
```

Comment:
idx >=0 here and in all similar places, after you added self_dim_size to idx it has to be non-negative.

Reply:
Sorry, I will swap these two `if`s